### ISSUE TITLE
Uncaught (in promise) CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 2f 2f 20 69

### ISSUE THREAD

**Issue by Coldthug** [emscripten-core/emscripten#20978]
Hello, I have encountered difficulties in learning this technology and come here to seek help.Currently a student with limited knowledge in programming.
I tried to run a simple example, and my steps and detailed code are presented below.
This is the CPP code：
```
#include <iostream>
#include <emscripten.h>

extern "C" {
    EMSCRIPTEN_KEEPALIVE
    int add(int a, int b) {
        return a + b;
    }
}
```

This is the HTML code:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>WebAssembly Addition</title>
</head>
<body>
    <h1>WebAssembly Addition</h1>
    <label for="num1">Enter the first number:</label>
    <input type="number" id="num1">

    <label for="num2">Enter the second number:</label>
    <input type="number" id="num2">

    <button onclick="performAddition()">Add</button>

    <div id="result"></div>

    <script>
        // 异步加载 WebAssembly 模块
        async function loadWebAssembly(filename) {
            const response = await fetch(filename);
            const bytes = await response.arrayBuffer();
            const result = await WebAssembly.instantiate(bytes, {});

            return result.instance;
        }

        // 在模块加载完成后执行的函数
        async function performAddition() {
            // 获取输入值
            const num1 = parseInt(document.getElementById('num1').value);
            const num2 = parseInt(document.getElementById('num2').value);

            // 加载 WebAssembly 模块
            const module = await loadWebAssembly('add.js');

            // 调用 C++ 函数
            const sum = module.exports.add(num1, num2);

            // 显示结果
            document.getElementById('result').innerText = `The sum of ${num1} and ${num2} is: ${sum}`;
        }
    </script>
</body>
</html>
```
Execute this command.The add.js and add.wasm files are generated.
```
em++ -o add.js add.cpp
```
Start local testing, execute this command, open the browser and access the website.
```
python3 -m http.server
```

The screenshot of the running result is as follows.
![image](https://github.com/emscripten-core/emscripten/assets/93389398/decd8eaa-5174-4874-bad2-f65339fa952a)

In response to this problem, I tried to search for a solution in the browser and looked at the previous problem feedback and comment areas.The official documents don’t seem to be friendly to newcomers, and I haven’t found a specific solution.Maybe it's because of my limited level.

Looking forward to your reply.Thank you!

**Comment by Coldthug** [comment]
@kripken 
Thank you for your answer. I have other questions that need to be answered.
There are two functional scenarios,Where can I get relevant knowledge or tutorials?
I couldn't find accurate and detailed usage documentation, so I asked Chatgpt.
------
1. Dynamic library .so
scene description：
There is a C/C++ program that uses a .so dynamic library without source code from outside, and compiles the program into a wasm application and print the content to the html page.
I learned about the following command, to compile using LLVM.
`llvm-dis -o myLibrary.ll -x86-64 myLibrary.so`
`emcc -o myLibrary.wasm myLibrary.ll`

2. nodeFS file system
scene description：
There is a text file hello.txt locally, and there is a C/C++ program that uses the nodeFS file system to read the text file. Compile the C/C++ program into a wasm application and print the text content to the html page.
How to use emcc command to generate related files.
-----

I checked out the following link：
https://www.cntofu.com/book/150/zh/ch3-runtime/ch3-03-fs.md
https://emscripten.org/docs/api_reference/Filesystem-API.html#filesystem-api-nodefs
https://wasmdev.cn/refrence-book/

The relevant books do not explain in detail how to use it.
This is not very beginner-friendly.

Looking forward to your answer again.Thank you.
